### PR TITLE
Adds release notes for FM dotnet 3.3.0

### PR DESCRIPTION
**Pull request by rossgrambo**

### CODE REVIEW

File: releaseNotes/Microsoft.Featuremanagement.md [hunk 1]
```diff
@@ -2,6 +2,17 @@
 
 [Source code][source_code] | [Package (NuGet)][package] | [Samples][samples] | [Product documentation][docs]
 
+## 3.3.0 - May 8, 2024
+
+### Enhancements
+
+* Added a `Recurrance` option to the `TimeWindow` filter. This allows a recurring period when the filter will be on. Now you can enable the flag every day after 5 PM or perhaps on Fridays from noon to 6 PM. ([#266](https://github.com/microsoft/FeatureManagement-Dotnet/pull/266)). See more details [here](https://github.com/microsoft/FeatureManagement-Dotnet?tab=readme-ov-file#recurrence-pattern).
```
Comment (jimmyca15) [inline releaseNotes/Microsoft.Featuremanagement.md:9]:
```suggestion
* Added a `Recurrence` option to the `TimeWindow` filter to support recurring time windows. This enables scenarios such as enabling the flag every day after five PM or enabling the flag on Fridays. ([#266](https://github.com/microsoft/FeatureManagement-Dotnet/pull/266)). See more details [here](https://github.com/microsoft/FeatureManagement-Dotnet?tab=readme-ov-file#recurrence-pattern).
```

File: releaseNotes/Microsoft.Featuremanagement.md [hunk 2]
```diff
@@ -2,6 +2,17 @@
...
+* Adjusted the built-in filters to optionally require a `LoggerFactory`. ([#386](https://github.com/microsoft/FeatureManagement-Dotnet/pull/386)).
```
Comment (jimmyca15) [inline releaseNotes/Microsoft.Featuremanagement.md:10]:
```suggestion
* Adjusted built-in filters to optionally require a `LoggerFactory`. ([#386](https://github.com/microsoft/FeatureManagement-Dotnet/pull/386)).
```

Comment (jimmyca15) [inline releaseNotes/Microsoft.Featuremanagement.md:10]: This is confusing me. Is it that we no longer require logger factory?

Comment (jimmyca15) [reply releaseNotes/Microsoft.Featuremanagement.md:10]: optionally required sounds like an oxymoron.

File: releaseNotes/Microsoft.Featuremanagement.md [hunk 3]
```diff
@@ -2,6 +2,17 @@
...
+
+### Bug fixes
+
+* Previously, refreshing while enumerating the async enumerable returned by `GetFeatureNamesAsync` could cause a null pointer exception. Now, enumeration will skip the flag if the definition is suddenly missing. ([#438](https://github.com/microsoft/FeatureManagement-Dotnet/pull/438))
```
Comment (jimmyca15) [inline releaseNotes/Microsoft.Featuremanagement.md:14]:
```suggestion
* Fixed a possible null-reference exception when enumerating `GetFeatureNamesAsync`. ([#438](https://github.com/microsoft/FeatureManagement-Dotnet/pull/438))
```

File: releaseNotes/Microsoft.Featuremanagement.md (hunk 2 above)
Comment (rossgrambo) [reply releaseNotes/Microsoft.Featuremanagement.md:10]: Yes. The built-in filters will default to null if no LoggerFactory is available. Do you prefer:

Made `LoggerFactory` an optional dependency for built-in filters

Comment (jimmyca15) [reply releaseNotes/Microsoft.Featuremanagement.md:10]: Reasonable, but dependency is vague. How about 

A `LoggerFactory` is no longer a required when constructing built-in filters.

Comment (rossgrambo) [reply releaseNotes/Microsoft.Featuremanagement.md]: Adjusted to

> A `LoggerFactory` is no longer required when constructing built-in filters.

File: releaseNotes/Microsoft.Featuremanagement.md [hunk 4]
```diff
@@ -2,6 +2,17 @@
 
 [Source code][source_code] | [Package (NuGet)][package] | [Samples][samples] | [Product documentation][docs]
 
+## 3.3.0 - May 8, 2024
+
+### Enhancements
+
+* Added a `Recurrence` option to the `TimeWindow` filter to support recurring time windows. This enables scenarios such as enabling the flag every day after five PM or enabling the flag on Fridays. See more details [here](https://github.com/microsoft/FeatureManagement-Dotnet?tab=readme-ov-file#recurrence-pattern). ([#266](https://github.com/microsoft/FeatureManagement-Dotnet/pull/266))
```
Comment (zhiyuanliang-ms) [inline releaseNotes/Microsoft.Featuremanagement.md:9]: Added a `Recurrence` option to the `TimeWindow` filter to support recurring time window. This enables scenarios where a feature flag is activated based on recurrence patterns, such as every day after 5 PM or every Friday.
```suggestion
* Added a `Recurrence` option to the `TimeWindow` filter to support recurring time window. This enables scenarios where a feature flag is activated based on recurrence patterns, such as every day after 5 PM or every Friday. See more details [here](https://github.com/microsoft/FeatureManagement-Dotnet?tab=readme-ov-file#recurrence-pattern). ([#266](https://github.com/microsoft/FeatureManagement-Dotnet/pull/266))
```

**Review by jimmyca15 — APPROVED** [review]

**Pull request merged**